common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/TraceTimeIntervalWarningService.java

```diff
@@ -22,46 +29,77 @@
+
   private final TraceTimeIntervalWarningRepository traceTimeIntervalWarningRepo;
+  private final FakeCheckinsGenerator fakeCheckinsGenerator;
 
   public TraceTimeIntervalWarningService(
-      TraceTimeIntervalWarningRepository traceTimeIntervalWarningRepo) {
+      TraceTimeIntervalWarningRepository traceTimeIntervalWarningRepo,
+      FakeCheckinsGenerator fakeCheckinsGenerator) {
     this.traceTimeIntervalWarningRepo = traceTimeIntervalWarningRepo;
+    this.fakeCheckinsGenerator = fakeCheckinsGenerator;
   }
 
   /**
    * Store the given checkin data as {@link TraceTimeIntervalWarning} entities. Returns the number
-   * of inserted entities useful for the case where there might be conflicts with the table
-   * constraints during the db save operations.
+   * of inserted entities, which is useful for the case where there might be conflicts with the
+   * table constraints during the db save operations.
    */
   @Transactional
-  public int saveCheckinData(List<CheckIn> checkins) {
+  public int saveCheckins(List<CheckIn> checkins) {
+    return saveCheckins(checkins, this::hashLocationId);
+  }
+
+  private int saveCheckins(List<CheckIn> checkins, Function<ByteString, byte[]> idHashGenerator) {
     int numberOfInsertedTraceWarnings = 0;
 
     for (CheckIn checkin : checkins) {
-      boolean traceWarningInsertedSuccessfully = traceTimeIntervalWarningRepo
-          .saveDoNothingOnConflict(checkin.getLocationId().toByteArray(),
-              checkin.getStartIntervalNumber(),
-              checkin.getEndIntervalNumber() - checkin.getStartIntervalNumber(),
-              checkin.getTransmissionRiskLevel(),
-              CheckinsDateSpecification.HOUR_SINCE_EPOCH_DERIVATION
-                  .apply(Instant.now().getEpochSecond()));
-
-      if (traceWarningInsertedSuccessfully) {
-        numberOfInsertedTraceWarnings++;
+      byte[] hashId = idHashGenerator.apply(checkin.getLocationId());
+      if (hashId != NO_HASH) {
+        boolean traceWarningInsertedSuccessfully = traceTimeIntervalWarningRepo
+            .saveDoNothingOnConflict(hashId,
+                checkin.getStartIntervalNumber(),
+                checkin.getEndIntervalNumber() - checkin.getStartIntervalNumber(),
+                checkin.getTransmissionRiskLevel(),
+                CheckinsDateSpecification.HOUR_SINCE_EPOCH_DERIVATION
+                    .apply(Instant.now().getEpochSecond()));
+
+        if (traceWarningInsertedSuccessfully) {
+          numberOfInsertedTraceWarnings++;
+        }
       }
     }
 
     int conflictingTraceWarnings = checkins.size() - numberOfInsertedTraceWarnings;
     if (conflictingTraceWarnings > 0) {
       logger.warn(
-          "{} out of {} TraceTimeIntervalWarnings conflicted with existing database entries and were ignored.",
+          "{} out of {} TraceTimeIntervalWarnings conflicted with existing "
+          + "database entries or had errors while storing "
+              + "and were ignored.",
```

Comment:
Would prefer to ignore on conflict, but reject the whole lot on error (as this might hint at a malicious, manufactured submission).

Reply:
Will be moved to the bean constructor and fail the app on startup in this scenario